portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +27,122 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  /**
+   *This welcome String will remain here for now as it is still 
+   *in one of my PRs and I am unsure as to what the best practice 
+   *is for these cases. In the meantime, this String will be sent as
+   *part of the JSON response.
+  */
+  private String welcome = "Welcome to my portfolio!";
+
+  // An array list containing all the data that will be recieved with a Get request.
+  private ArrayList<Object[]> data;
+  
+  /**
+   * A linked list that will store every comment on the site.
+   *
+   * A linked list is used instead of an ArrayList in order to ensure that a deletion 
+   * implementation occurs faster in cases where the comment to be deleted is in the
+   * middle of the list.
+   */
+  private LinkedList<Comment> comments = new LinkedList<Comment>();;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // Initialize Linked List/s
+    data = new ArrayList<Object[]>();
```

Comment:
If possible, I'd recommend using a custom object to hold this data and then using Gson to convert it to JSON. Having a list of generic object arrays is error prone since it's possible for any type of object to be in the list which can result in unexpected behavior. It's also more legible when you have a class that defines the structure.